cmake/Dependencies.cmake

```diff
@@ -10,21 +18,30 @@ list(APPEND Caffe_LINKER_LIBS ${Boost_LIBRARIES})
 find_package(Threads REQUIRED)
 list(APPEND Caffe_LINKER_LIBS ${CMAKE_THREAD_LIBS_INIT})
 
-# ---[ Google-glog
-include("cmake/External/glog.cmake")
-include_directories(SYSTEM ${GLOG_INCLUDE_DIRS})
-list(APPEND Caffe_LINKER_LIBS ${GLOG_LIBRARIES})
-
 # ---[ Google-gflags
 include("cmake/External/gflags.cmake")
 include_directories(SYSTEM ${GFLAGS_INCLUDE_DIRS})
 list(APPEND Caffe_LINKER_LIBS ${GFLAGS_LIBRARIES})
 
+# ---[ Google-glog
+if(MSVC)
+    add_definitions(-DGOOGLE_GLOG_DLL_DECL=)
+endif()
+include("cmake/External/glog.cmake")
```

Comment:
I think this should be done only if user explicitly set special option for this.